pkg/validation/internal/operatorhub.go

```diff
@@ -0,0 +1,194 @@
+		}
+	}
+
+	if csv.Spec.Icon != nil {
+		// only one icon is allowed
+		if len(csv.Spec.Icon) != 1 {
+			errs = append(errs, fmt.Errorf("csv.Spec.Icon should only have one element"))
+		}
+
+		for _, icon := range csv.Spec.Icon {
+			if icon.MediaType == "" || icon.Data == "" {
+				errs = append(errs, fmt.Errorf("csv.Spec.Icon elements should contain both data and mediatype"))
+			}
+
+			if icon.MediaType != "" {
+				validMediatypes := map[string]struct{}{
+					"image/gif":     struct{}{},
+					"image/jpeg":    struct{}{},
+					"image/png":     struct{}{},
+					"image/svg+xml": struct{}{},
+				}
+
+				if _, ok := validMediatypes[icon.MediaType]; !ok {
+					errs = append(errs, fmt.Errorf("csv.Spec.Icon %s does not have a valid mediatype", icon.MediaType))
+				}
+			}
+		}
+	} else {
+		errs = append(errs, fmt.Errorf("csv.Spec.Icon not specified"))
+	}
+
+	if categories, ok := csv.ObjectMeta.Annotations["categories"]; ok {
+		categorySlice := strings.Split(categories, ",")
+
+		validCategories := map[string]struct{}{
+			"AI/Machine Learning":    struct{}{},
+			"Application Runtime":    struct{}{},
+			"Big Data":               struct{}{},
+			"Cloud Provider":         struct{}{},
+			"Developer Tools":        struct{}{},
+			"Database":               struct{}{},
+			"Integration & Delivery": struct{}{},
+			"Logging & Tracing":      struct{}{},
+			"Monitoring":             struct{}{},
+			"Networking":             struct{}{},
+			"OpenShift Optional":     struct{}{},
+			"Security":               struct{}{},
+			"Storage":                struct{}{},
+			"Streaming & Messaging":  struct{}{},
+		}
+
+		for _, category := range categorySlice {
+			if _, ok := validCategories[category]; !ok {
+				errs = append(errs, fmt.Errorf("csv.Metadata.Annotations.Categories %s is not a valid category", category))
+			}
+		}
+	}
+
+	return errs
+}
+
+func checkAlmExamples(csv v1alpha1.ClusterServiceVersion) []error {
+	errs := make([]error, 0)
+
+	annotations := csv.ObjectMeta.Annotations
+
+	// Expect either `alm-examples` or `olm.examples` but not both
+	var examplesString string
+	almExamples, almOK := annotations["alm-examples"]
+	olmExamples, olmOK := annotations["olm.examples"]
+	if almOK {
+		if olmOK {
+			errs = append(errs, fmt.Errorf("both `csv.metadata.annotations.alm-examples` and `csv.metadata.annotations.olm.examples` are present"))
+		}
+		examplesString = almExamples
+	} else {
+		examplesString = olmExamples
+	}
+
+	// parse alm examples
+	examples := []unstructured.Unstructured{}
+	dec := yaml.NewYAMLOrJSONDecoder(strings.NewReader(examplesString), 8)
+	if err := dec.Decode(&examples); err != nil && err != io.EOF {
+		errs = append(errs, errors.ErrInvalidParse("error decoding example CustomResource", err))
+		return errs
+	}
+	parsed := map[string]struct{}{}
+	for _, example := range examples {
+		parsed[example.GetObjectKind().GroupVersionKind().Kind] = struct{}{}
```

Comment:
What happens if we have multiple items with a same `Kind`? Is that case possible because if it is, then key collision will happen here.